code/game/objects/effects/spawners/minor_variation.dm

```diff
@@ -0,0 +1,58 @@
+/*
+Makes variations possible with code only.
+This is mainly to prevent creating new map files when creating smaller variations that aren't worth a map file. - Hopek
+*/
+
+/obj/effect/variation
+	name = "variation"
+	icon = 'icons/effects/landmarks_variation.dmi'
+	icon_state = "generic_green"
+	var/variations // number of variations
+	var/choice // chosen variation by choice()
+	var/chosen // has a choice been made for this place?
+	var/location_place // location of placement calculated from reference
+	var/ref_x // x of reference
+	var/ref_y // y of reference
+	var/ref_z // z of reference
+
+/obj/effect/variation/Initialize()
+	. = ..()
+	chosen = FALSE
+
+/obj/effect/variation/proc/place(x, y , item , direction)
+	if(!x)
```

Comment:
if you change the defaults then there's no need for these, (it's your own fault if you put in null as the coordinate)

Reply:
Well yeah but I don't really trust people not to do something weird. 
Some time years in the future, someone is going to mess up and cause runtimes.
I'll did the x=0,y=0 to satisfy your need to make changes to my PR though.

Reply:
Yeah but this is one of those things that you'd have to do REALLY wrong, like runtime worthy wrong
I'd rather want a runtime at that point so we don't get weird stuff where stuff starts spawning at the relative 0/0